src/ClientData/ScopeCollectionTest.cpp

```diff
@@ -56,43 +59,45 @@ static void AssertStatsAreEqual(ScopeStats actual, ScopeStats expect) {
 TEST(ScopeCollectionTest, CreateEmpty) {
   ScopeCollection collection = ScopeCollection();
   ASSERT_TRUE(collection.GetAllProvidedScopeIds().empty());
-  ScopeStats stats = collection.GetScopeStatsOrDefault(kScopeId);
+  ScopeStats stats = collection.GetScopeStatsOrDefault(kScopeId1);
   AssertStatsAreEqual(stats, kDefaultScopeStats);
-  ASSERT_EQ(collection.GetSortedTimerDurationsForScopeId(kScopeId), nullptr);
+  ASSERT_EQ(collection.GetSortedTimerDurationsForScopeId(kScopeId1), nullptr);
 }
 
 TEST(ScopeCollectionTest, AddTimersWithUpdateStats) {
   ScopeCollection collection = ScopeCollection();
   for (TimerInfo timer : kTimersScopeId1) {
-    collection.UpdateScopeStats(kScopeId, timer);
+    collection.UpdateScopeStats(kScopeId1, timer);
   }
   ASSERT_EQ(collection.GetAllProvidedScopeIds().size(), 1);
 
-  collection.UpdateScopeStats(ScopeId(2), kTimerScopeId2);
+  collection.UpdateScopeStats(kScopeId2, kTimerScopeId2);
   ASSERT_EQ(collection.GetAllProvidedScopeIds().size(), 2);
 
-  AssertStatsAreEqual(collection.GetScopeStatsOrDefault(kScopeId), kScope1Stats);
-  const auto* timer_durations = collection.GetSortedTimerDurationsForScopeId(kScopeId);
+  AssertStatsAreEqual(collection.GetScopeStatsOrDefault(kScopeId1), kScope1Stats);
+  const auto* timer_durations = collection.GetSortedTimerDurationsForScopeId(kScopeId1);
   ASSERT_NE(timer_durations, nullptr);
   ASSERT_EQ(timer_durations->size(), kNumTimers);
   ASSERT_THAT(*timer_durations, testing::ElementsAre(9, 500, 3000));
 }
 
 TEST(ScopeCollectionTest, CreateWithTimers) {
-  orbit_grpc_protos::CaptureOptions capture_options;
-  std::unique_ptr<ScopeIdProvider> scope_id_provider =
-      NameEqualityScopeIdProvider::Create(capture_options);
+  MockScopeIdProvider mock_scope_id_provider;
   std::vector<const TimerInfo*> timers;
+  timers.push_back(&kTimerScopeId2);
   for (size_t i = 0; i < kNumTimers; ++i) {
     const TimerInfo* timer = &kTimersScopeId1.at(i);
-    timers.emplace_back(timer);
+    timers.push_back(timer);
   }
-  timers.emplace_back(&kTimerScopeId2);
-  ScopeCollection collection = ScopeCollection(*scope_id_provider, timers);
+  EXPECT_CALL(mock_scope_id_provider, ProvideId(testing::_))
+      .Times(4)
+      .WillOnce(testing::Return(kScopeId2))
+      .WillRepeatedly(testing::Return(kScopeId1));
```

Comment:
I think, `ProvideId(testing::_)` can be replaced with just `ProvideId`, as you don't match any arguments. Also, I suggest to add `using ::testing::Return` and `testing::ElementsAre` to improve readability.

Reply:
Done